agent.go

```diff
@@ -1288,11 +1291,77 @@ func cgroupsMount() error {
 	return ioutil.WriteFile(cgroupMemoryUseHierarchyPath, []byte{'1'}, cgroupMemoryUseHierarchyMode)
 }
 
+func setupDebugConsoleForVsock(ctx context.Context) error {
+	var shellPath string
+	for _, s := range supportedShells {
+		var err error
+		if _, err = os.Stat(s); err == nil {
+			shellPath = s
+			break
+		}
+		agentLog.WithError(err).WithField("shell", s).Warn("Shell not found")
+	}
+
+	if shellPath == "" {
+		return fmt.Errorf("No available shells (checked %v)", supportedShells)
+	}
+
+	cmd := exec.Command(shellPath, "-i")
+	cmd.Env = os.Environ()
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		// Create Session
+		Setsid: true,
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				// stop the thread
+				return
+			default:
+				dcmd := *cmd
+
+				l, err := vsock.Listen(debugConsoleVSockPort)
+				if err != nil {
+					// nobody dialing
+					time.Sleep(time.Second)
```

Comment:
ooi, why do we need the sleep (and why 1 second)? :)

Reply:
good point, these sleeps are not needed